mesh/amqp.h

```diff
@@ -21,15 +21,27 @@
 
 struct mesh_amqp;
 
+struct mesh_amqp_config {
+	const char *url;
+	const char *exchange;
+	const char *routing_key;
+};
+
 struct mesh_amqp *mesh_amqp_new(void);
 void mesh_amqp_free(struct mesh_amqp *amqp);
 
 const char *mesh_amqp_get_url(struct mesh_amqp *amqp);
-bool mesh_amqp_set_url(struct mesh_amqp *amqp, const char *url);
+void mesh_amqp_set_url(struct mesh_amqp *amqp, const char *url);
+
 const char *mesh_amqp_get_exchange(struct mesh_amqp *amqp);
```

Comment:
If this is supposed to be `free()d` by the caller, then return a non-const pointer. This way you won't need to cast when calling `l_free()`